Data/ByteString.hs

```diff
@@ -1,6 +1,7 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE MagicHash, UnboxedTuples,
-            NamedFieldPuns, BangPatterns #-}
+            NamedFieldPuns, BangPatterns,
+            ScopedTypeVariables #-}
```

Comment:
We can avoid the need for `ScopedTypeVariables` if we use a type annotation on a value rather than in a pattern. Eg.

      filesz <- catch (hFileSize h) useZeroIfNotRegularFile
      ...
    where
      useZeroIfNotRegularFile :: IOException -> IO Int
      useZeroIfNotRegularFile _ = return 0